lib/internal/http2/core.js

```diff
@@ -868,6 +868,7 @@ class Http2Session extends EventEmitter {
 
     if (!socket._handle || !socket._handle._externalStream) {
       socket = new StreamWrap(socket);
+      socket.on('error', () => {});
```

Comment:
I don't think we should be swallowing the error on these. Those should be forwarded on so they can be properly handled.

Reply:
@jasnell agreed. Should I `throw` something descriptive inside the function block? Just forwarding the same error further seems counterintuitive.

Perhaps something that'd make the failure (and why exactly it happened) more obvious?

Reply:
@mafintosh and @mcollina ... what do you think?

Reply:
I think the error should be forwarded to `session`.

Reply:
@mcollina 

by `session`, I suppose you mean the current instance of `Http2Session` itself, right? I inferred that from `socket[kSession] = this`

If that's the case, I hope the following code does what you expected:

```diff
if (!socket._handle || !socket._handle._externalStream) {
       socket = new StreamWrap(socket);
-      socket.on('error', () => {});
+      socket.on('error', (err) => {
+        this.emit('error', err);
+      });
     }
```